cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20AuthorizeController.java

```diff
@@ -54,6 +55,8 @@ public ModelAndView handleRequestInternal(final HttpServletRequest request, fina
         logger.debug("bypassApprovalParameter: {}", bypassApprovalParameter);
 
         final OAuthRegisteredService registeredService = OAuthUtils.getRegisteredOAuthService(this.servicesManager, clientId);
+        RegisteredServiceAccessStrategySupport.ensureServiceAccessIsAllowed(clientId, registeredService);
```

Comment:
Don't we want to handle that properly instead of throwing an exception and generating a 500 error?